### PR TITLE
Redesign Action Items & ActionCategories, Shift AgendaCategory to Settings & Fix LeftDrawer Scroll

### LINKED ISSUE #2165: GSoC: Redesign & Add Delete, Search, Sort for Action Item Categories (ADMIN)
**Is your feature request related to a problem? Please describe.**
There are various common features such as search sort and delete features are missing on Action Items & Action item Category Screen. These screens could be redesigned and refactored to match UI system of Talawa Project
Bug: 
- Organization Left Drawer misses scroll bar, so for small height screens clicking on last options is not possible
- In Left Drawer, Organization image avatar gets stretched on small screens 

**Describe the solution you'd like**

**Action Item Category Screen:**
- Add Delete functionality (Modal) & description while creation
- Add Search, search by and Sort functionality
- Display Active/Disabled for each Action Item Categories 
- Redesign screen, Add/View & Edit Modal for Action Item Category

**Action Item Screen (Global):**
- Add search options along with filters
- Add Delete Functionality (Modal)
- Add Due Date & status as Late or Label like 2 days remaining per action item
- Redesign screen, Add/View & Edit Modal for Action Item Category

- Similarly update Event Dashboard Event Actions screen w.r.t global *Action Items Screen.

Existing Screens:
Action Item Category (Settings)
image

Action Items screen (Global)
image

Action Items screen in Event Management Dashboard
image

### PR DESCRIPTION
**Pull request by GlenDsza**
**What kind of change does this PR introduce?**

bugfix, feature, refactoring

**Issue Number:** 

Fixes #2165, #2223 & #2225 

**Did you add tests for your changes?**

Yes 

**Snapshots/Videos:**

https://github.com/user-attachments/assets/10b4b6af-bb6c-4d2f-9acf-9e25f9bdc21a

**Summary**

- Refactored General Settings Page (Layout).
- Redesigned & Refac Action Items Categories Screen/Modals, added search, sort & filtering options.
- Redesigned & Refac Action Items Screen/Modals, added searchBy, sort & filtering options.
- Added Scroll bar in Organization Left Drawer & few CSS changes to ensure last tab is accessible and not hidden below Drawer Collapse Button.
- Shifted Agenda Category from Left Drawer to Settings Tab.
- Removed unused SVG files  

**Does this PR introduce a breaking change?**

Yes, this should only be merged once [API PR](https://github.com/PalisadoesFoundation/talawa-api/issues/2492) is done

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes 

## Summary by CodeRabbit

- **New Features**
	- Enhanced localization support with added translations for various statuses and actions in multiple languages (English, French, Hindi, Spanish, Chinese).
	- Introduced new modal components for creating and updating action item categories and action items.
	- Added functionality for managing action item categories, including filtering and sorting options.
	- Implemented a new dropdown menu for mobile views in the Event Management component.

- **Bug Fixes**
	- Corrected error messages and improved clarity in user notifications.
	- Fixed issues related to the rendering of certain components and their states.

- **Documentation**
	- Updated test cases to improve coverage and ensure accurate functionality of new components.

- **Style**
	- Enhanced UI styling for various components, improving overall user experience and responsiveness.

- **Tests**
	- Introduced comprehensive unit tests for new modal components and existing functionalities to ensure reliability and maintainability.

### CODE REVIEW

**Comment by GlenDsza** [comment]
For now the introspection check is expected to fail, It will be fixed in PR for [API issue](https://github.com/PalisadoesFoundation/talawa-api/issues/2492), Once its fixed then we can proceed this PR

**Comment by palisadoes** [comment]
1. Please increase the code coverage for the patch.
2. Should this PR be merged before the one for the API issue?

**Comment by GlenDsza** [comment]
> 1. Please increase the code coverage for the patch.
> 2. Should this PR be merged before the one for the API issue?
> 3. Please follow the suggestions by coderabbit

Should be merged after the API

File: src/components/OrgSettings/ActionItemCategories/OrgActionItemCategories.module.css [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+.active {
+  background-color: #31bb6a50 !important;
+}
+
+.pending {
+  background-color: #ffd76950 !important;
+  color: #bb952bd0 !important;
+  border-color: #bb952bd0 !important;
+}
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/components/OrgSettings/ActionItemCategories/OrgActionItemCategories.module.css:117]: Added

File: src/screens/OrganizationActionItems/ItemDeleteModal.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          <Button variant="secondary" onClick={hide} data-testid="deletenobtn">
+            {tCommon('no')}
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    </>
+  );
+};
+
+export default ItemDeleteModal;
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationActionItems/ItemDeleteModal.tsx:92]: Tests for ItemDeleteModal are already written in ItemDeleteModal.test.tsx with 100% coverage

File: src/screens/OrgSettings/OrgSettings.test.tsx [hunk 3]
```diff
@@ -1,171 +1,107 @@
...
+                  element={<div data-testid="paramsError"></div>}
+                />
+              </Routes>
+            </I18nextProvider>
+          </LocalizationProvider>
+        </Provider>
+      </MemoryRouter>
+    </MockedProvider>,
+  );
+};
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrgSettings/OrgSettings.test.tsx:42]: Not necessary

File: src/screens/OrganizationActionItems/ItemDeleteModal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        <BrowserRouter>
+          <LocalizationProvider dateAdapter={AdapterDayjs}>
+            <I18nextProvider i18n={i18nForTest}>
+              <ItemDeleteModal {...props} />
+            </I18nextProvider>
+          </LocalizationProvider>
+        </BrowserRouter>
+      </Provider>
+    </MockedProvider>,
+  );
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationActionItems/ItemDeleteModal.test.tsx:89]: Adding comments in test files is not necessary for render functions

File: src/screens/OrgSettings/OrgSettings.mocks.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                __typename: 'User',
+              },
+            ],
+            membershipRequests: [],
+            blockedUsers: [],
+            __typename: 'Organization',
+          },
+        ],
+      },
+    },
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrgSettings/OrgSettings.mocks.ts:70]: All the scenarios with/without image and all sorting and searching scenarios are covered

File: src/screens/OrgSettings/OrgSettings.mocks.ts [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+      data: {
+        customFieldsByOrganization: [
+          {
+            _id: 'adsdasdsa334343yiu423434',
+            type: 'fieldType',
+            name: 'fieldName',
+          },
+        ],
+      },
+    },
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrgSettings/OrgSettings.mocks.ts:87]: Above test is enough for achieving 100% test coverage

File: src/components/OrgSettings/ActionItemCategories/CategoryModal.tsx [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+import { useTranslation } from 'react-i18next';
+import type { InterfaceActionItemCategoryInfo } from 'utils/interfaces';
+import { useMutation } from '@apollo/client';
+import {
+  CREATE_ACTION_ITEM_CATEGORY_MUTATION,
+  UPDATE_ACTION_ITEM_CATEGORY_MUTATION,
+} from 'GraphQl/Mutations/ActionItemCategoryMutations';
+import { toast } from 'react-toastify';
+import { FormControl, TextField } from '@mui/material';
+
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/components/OrgSettings/ActionItemCategories/CategoryModal.tsx:13]: Components from both libraries are required as they are used as they provide certain option and inbuilt styles

File: src/components/OrgSettings/ActionItemCategories/CategoryModal.tsx [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+            data-testid="formSubmitButton"
+          >
+            {mode === 'create'
+              ? tCommon('create')
+              : t('updateActionItemCategory')}
+          </Button>
+        </Form>
+      </Modal.Body>
+    </Modal>
+  );
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/components/OrgSettings/ActionItemCategories/CategoryModal.tsx:204]: Components from both libraries are required as they are used as they provide certain option and inbuilt styles

File: src/screens/EventManagement/EventManagement.tsx [hunk 9]
```diff
@@ -154,65 +145,99 @@ const EventManagement = (): JSX.Element => {
...
+              </div>
+            );
+          case 'eventStats':
+            return (
+              <div data-testid="eventStatsTab">
+                <h2>Event Statistics</h2>
+              </div>
+            );
+        }
+      })()}
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/EventManagement/EventManagement.tsx:240]: abstracted for onClick handlers for navigation in separate function

File: src/components/OrgSettings/ActionItemCategories/CategoryModal.test.tsx [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+  it('Try to edit without changing any field', async () => {
+    renderCategoryModal(link1, categoryProps[1]);
+    const submitBtn = screen.getByTestId('formSubmitButton');
+    userEvent.click(submitBtn);
+
+    await waitFor(() => {
+      expect(toast.error).toHaveBeenCalledWith(translations.sameNameConflict);
+    });
+  });
+});
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/components/OrgSettings/ActionItemCategories/CategoryModal.test.tsx:225]: Abstracted the common rendering logic for filling form and submitting in common function

File: src/components/OrgSettings/ActionItemCategories/OrgActionItemCategories.test.tsx [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+    });
+  });
+
+  it('should render the Action Item Categories Screen with error', async () => {
+    renderActionItemCategories(link3, 'orgId');
+    await waitFor(() => {
+      expect(screen.getByTestId('errorMsg')).toBeInTheDocument();
+    });
+  });
+});
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/components/OrgSettings/ActionItemCategories/OrgActionItemCategories.test.tsx:241]: There isn't much shared code which could be abstracted in separate function

File: src/screens/OrganizationActionItems/OrganizationActionItem.mocks.ts [hunk 12]
```diff
@@ -0,0 +1,482 @@
...
+    },
+    result: {
+      data: {
+        actionItemsByOrganization: [],
+      },
+    },
+  },
+  memberListQuery,
+  actionItemCategoryListQuery,
+];
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationActionItems/OrganizationActionItem.mocks.ts:482]: Above mocks are enough to test almost all possible scenarios

File: public/locales/fr/translation.json [hunk 13]
```diff
@@ -311,7 +310,14 @@
...
     "successfulDeletion": "Élément d'action supprimé avec succès",
-    "title": "Éléments d'action"
+    "title": "Éléments d'action",
+    "category": "Category",
+    "allotedHours": "Heures allouées",
+    "latestDueDate": "Date d'échéance la plus récente",
+    "earliestDueDate": "Date d'échéance la plus ancienne",
+    "updateActionItem": "Mettre à jour l'élément d'action",
+    "noneUpdated": "Aucun des champs n'a été mis à jour",
+    "updateStatusMsg": "Êtes-vous sûr de vouloir marquer cet élément d'action comme en attente?"
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply public/locales/fr/translation.json:320]: Updated

File: src/screens/OrganizationActionItems/ItemViewModal.test.tsx [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+    expect(screen.getByTestId('Wilt_avatar')).toBeInTheDocument();
+  });
+
+  it('should render ItemViewModal with completed item & assignee with null image', () => {
+    renderItemViewModal(link1, itemProps[1]);
+    expect(screen.getByText(t.actionItemDetails)).toBeInTheDocument();
+    expect(screen.getByTestId('Jane_image')).toBeInTheDocument();
+    expect(screen.getByTestId('Wilt_image')).toBeInTheDocument();
+  });
+});
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationActionItems/ItemViewModal.test.tsx:139]: Added more assertions for conditional rendering

File: public/locales/hi/common.json [hunk 15]
```diff
@@ -82,5 +82,14 @@
...
+  "successfullyUpdated": "सफलतापूर्वक अपडेट किया गया",
+  "all": "सभी",
+  "active": "सक्रिय",
+  "disabled": "अक्षम",
+  "pending": "लंबित",
+  "completed": "पूरा हुआ",
+  "late": "देर से",
+  "createdLatest": "सबसे पहले बनाया",
+  "createdEarliest": "सबसे जल्दी बनाया",
+  "searchBy": "के द्वारा खोजें {{item}}"
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply public/locales/hi/common.json:94]: Updated translations

**Comment by tasneemkoushar** [comment]
@palisadoes this PR is reviewed. didn't approved since waiting on the api-pr

**Comment by tasneemkoushar** [comment]
@GlenDsza do let me know when you are ready with your API pr

**Comment by tasneemkoushar** [comment]
@GlenDsza till then please convert this pr to draft pr

File: src/screens/OrganizationActionItems/ItemModal.tsx [hunk 16]
```diff
@@ -0,0 +1,477 @@
...
+      setActionItemCategory(null);
+      setAssignee(null);
+
+      actionItemsRefetch();
+      hide();
+      toast.success(t('successfulCreation'));
+    } catch (error: unknown) {
+      toast.error((error as Error).message);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationActionItems/ItemModal.tsx]: Done

**Review by tasneemkoushar — APPROVED** [review]

File: src/components/OrgSettings/ActionItemCategories/CategoryModal.test.tsx [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/components/OrgSettings/ActionItemCategories/CategoryModal.test.tsx:208]: Most of the code is already optimized, shifting just a line to beforeEach doesn't make sense. And even we need to pass different staticMock links for different tests

File: src/screens/ForgotPassword/ForgotPassword.tsx [hunk 18]
```diff
@@ -90,20 +90,16 @@ const ForgotPassword = (): JSX.Element => {
...
-          toast.error(tErrors('talawaApiUnavailable') as string);
-        } else {
-          toast.error(tErrors('errorSendingMail') as string);
-        }
+      if ((error as Error).message === 'User not found') {
+        toast.warn(tErrors('emailNotRegistered'));
+      } else if ((error as Error).message === 'Failed to fetch') {
+        toast.error(tErrors('talawaApiUnavailable'));
+      } else {
+        toast.error(tErrors('errorSendingMail'));
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/ForgotPassword/ForgotPassword.tsx:102]: If error occurs, the catch block catches its, rest in other cases data is always returned with otpToken inside

File: src/screens/OrganizationActionItems/ItemViewModal.test.tsx [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+    expect(screen.getByText(t.actionItemDetails)).toBeInTheDocument();
+    expect(screen.getByTestId('Jane_image')).toBeInTheDocument();
+    expect(screen.getByTestId('Wilt_image')).toBeInTheDocument();
+    expect(
+      screen.queryByLabelText(t.postCompletionNotes),
+    ).not.toBeInTheDocument();
+    expect(screen.getByLabelText(t.allotedHours)).toBeInTheDocument();
+    expect(screen.getByLabelText(t.allotedHours)).toHaveValue('-');
+  });
+});
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationActionItems/ItemViewModal.test.tsx:147]: The code which is conditional rendered such as allotedHours is checked

File: public/locales/fr/translation.json [hunk 20]
```diff
@@ -311,7 +310,14 @@
...
     "successfulDeletion": "Élément d'action supprimé avec succès",
-    "title": "Éléments d'action"
+    "title": "Éléments d'action",
+    "category": "Catégorie",
+    "allotedHours": "Heures allouées",
+    "latestDueDate": "Date d'échéance la plus récente",
+    "earliestDueDate": "Date d'échéance la plus ancienne",
+    "updateActionItem": "Mettre à jour l'élément d'action",
+    "noneUpdated": "Aucun des champs n'a été mis à jour",
+    "updateStatusMsg": "Êtes-vous sûr de vouloir marquer cet élément d'action comme en attente?"
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply public/locales/fr/translation.json:320]: The change form category to Categorie is already done

File: src/screens/OrganizationActionItems/ItemModal.tsx [hunk 21]
```diff
@@ -0,0 +1,440 @@
...
+            className={styles.greenregbtn}
+            data-testid="submitBtn"
+          >
+            {editMode ? t('updateActionItem') : t('createActionItem')}
+          </Button>
+        </Form>
+      </Modal.Body>
+    </Modal>
+  );
+};
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationActionItems/ItemModal.tsx:438]: The numeric fields like allotedHours are checked at client side, rest input fields for text can contain anything numeric/symbols/text. And these fields like notes and allotedHours are optional so not checking for empty. And error handling for create /update is already present

**Pull request merged**